flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTypeUtil.java

```diff
@@ -132,4 +133,45 @@ private static TypeInformation toFlinkPrimitiveType(PrimitiveTypeInfo hiveType)
 					String.format("Flink doesn't support Hive primitive type %s yet", hiveType));
 		}
 	}
+
+	/**
+	 * Converts a Flink {@link TypeInformation} to corresponding Hive {@link TypeInfo}.
+	 */
+	public static TypeInfo toHiveTypeInfo(TypeInformation flinkType) {
```

Comment:
can we try to unify `toHiveTypeInfo()` and `toHiveType()` so we have only one path to covert Flink type to Hive type?

Reply:
I'll use Hive's `TypeInfoFactory` to achieve this. Alternatively, we can maintain a mapping from type name to `TypeInfo` ourselves.